python/katana_setup.py

```diff
@@ -184,27 +183,37 @@ def check_cython_module(name, cython_code, python_code="", extension_options=Non
         requirement_cache.add(cython_code, python_code, extension_options)
 
 
+def parse_text(fi):
+    result = {}
+    for line in fi:
+        line = line.strip()
+        if not line:
+            continue
+        idx = line.index("=")
+        key = line[:idx]
+        value = line[idx + 1 :]
+        result[key] = value
```

Comment:
I think you can just write `key, value = line.strip().split("=", maxsplit=1)` or something similar and avoid all the indexing mess.